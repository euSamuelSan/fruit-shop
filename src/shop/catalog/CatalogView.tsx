import React from 'react'
import styled from 'styled-components'
import Colors from '../../commons/Colors'
import SearchBar from '../../commons/components/SearchBar'
import ProductsGrid from '../productsGrid'

export type CatalogViewProps = {
    search: string
    handleOnChangeSearch: (e: React.ChangeEvent<HTMLInputElement>) => void
}

export default function CatalogView({
    search,
    handleOnChangeSearch,
}: CatalogViewProps): JSX.Element {
    return (
        <Container>
            <CatalogHeader>
                <SearchBar
                    placeholder="Pesquise por nome do produto"
                    value={search}
                    onChange={handleOnChangeSearch}
                />
            </CatalogHeader>
            <ProductsGrid searchTerm={search} />
        </Container>
    )
}

const Container = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    background-color: ${Colors.veryLightPink};
    min-height: 100vh;
    padding: 10px 0;
`

const CatalogHeader = styled.div`
    display: flex;
    padding: 10px 0;
`
