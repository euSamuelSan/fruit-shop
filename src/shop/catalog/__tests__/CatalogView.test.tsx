import { render, RenderResult, fireEvent, screen } from '@testing-library/react'

import CatalogView, { CatalogViewProps } from '../CatalogView'

describe('CatalogView', () => {
    const getProps = (): CatalogViewProps => ({
        search: '',
        handleOnChangeSearch: jest.fn(),
    })

    const renderCatalog = (props = getProps()): RenderResult => {
        return render(<CatalogView {...props} />)
    }

    it('Deve desenhar o componente corretamente', () => {
        renderCatalog()

        expect(document.body).toMatchSnapshot()
    })

    it('Deve chamar o handleOnChangeSearch ao alterar o valor do input', () => {
        const props = getProps()
        renderCatalog(props)

        const input = screen.getByPlaceholderText(
            'Pesquise por nome do produto'
        )
        fireEvent.change(input, { target: { value: 'a' } })

        expect(props.handleOnChangeSearch).toHaveBeenCalled()
    })
})
