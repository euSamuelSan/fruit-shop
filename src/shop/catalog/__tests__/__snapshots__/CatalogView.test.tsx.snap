// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogView Deve desenhar o componente corretamente 1`] = `
<body>
  <div>
    <div
      class="sc-breuTD gRSwPg"
    >
      <div
        class="sc-ksZaOG hWsKsx"
      >
        <div
          class="sc-gsnTZi oaevL"
        >
          <svg
            class="sc-dkzDqf gavTNQ"
            color="#8e8e93"
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(142, 142, 147);"
            viewBox="0 0 1024 1024"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
            />
          </svg>
          <input
            class="sc-bczRLJ hKuxkL sc-hKMtZM bTFJgM"
            placeholder="Pesquise por nome do produto"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="sc-iBkjds hMNVUq"
      />
    </div>
  </div>
</body>
`;
